cpp/models/sde_sir/simulation.h

```diff
@@ -0,0 +1,119 @@
+*
+* Authors: Nils Wassmuth, Rene Schmieding, Martin J. Kuehn
+*
+* Contact: Martin J. Kuehn <Martin.Kuehn@DLR.de>
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+#ifndef MIO_SDE_SIR_SIMULATION_H
+#define MIO_SDE_SIR_SIMULATION_H
+
+#include "memilio/compartments/flow_simulation.h"
+#include "memilio/compartments/simulation.h"
+#include "sde_sir/model.h"
+
+namespace mio
+{
+namespace ssir
+{
+
+/// @brief A specialized Simulation for mio::ssir::Model.
+class Simulation : public mio::Simulation<Model>
+{
+public:
+    using mio::Simulation<Model>::Simulation;
+    /**
+     * @brief advance simulation to tmax
+     * tmax must be greater than get_result().get_last_time_point()
+     * @param tmax next stopping point of simulation
+     */
+    Eigen::Ref<Eigen::VectorXd> advance(double tmax)
+    {
+        return get_ode_integrator().advance(
+            [this](auto&& y, auto&& t, auto&& dydt) {
+                get_model().step_size = get_dt();
+                get_model().get_derivatives(y, y, t, dydt);
+            },
+            tmax, get_dt(), get_result());
+    }
+};
+
+/**
+ * @brief Run a Simulation of a mio::ssir::Model.
+ * @param[in] t0 Start time.
+ * @param[in] tmax End time.
+ * @param[in] dt Initial step size of integration.
+ * @param[in] model An instance of mio::ssir::Model.
+ * @param[in] integrator Optionally override the IntegratorCore used by the Simulation.
+ * @return A TimeSeries to represent the final simulation result
+ */
+inline TimeSeries<ScalarType> simulate(double t0, double tmax, double dt, Model const& model,
+                                       std::shared_ptr<IntegratorCore> integrator = nullptr)
+{
+    return mio::simulate<Model, Simulation>(t0, tmax, dt, model, integrator);
+}
+
+/// @brief A specialized FlowSimulation for mio::ssir::Model.
+class FlowSimulation : public mio::FlowSimulation<Model>
+{
+public:
+    using mio::FlowSimulation<Model>::FlowSimulation;
+    /**
+     * @brief Advance the simulation to tmax.
+     * tmax must be greater than get_result().get_last_time_point().
+     * @param[in] tmax Next stopping time of the simulation.
+     */
+    Eigen::Ref<Eigen::VectorXd> advance_stochastic(double tmax)
+    {
+        assert(m_flow_result.get_num_time_points() == this->get_result().get_num_time_points());
+        auto result = this->get_ode_integrator().advance(
+            // see the general mio::FlowSimulation for more details on this DerifFunktion
```

Comment:
```suggestion
            // see the general mio::FlowSimulation for more details on this derived function.
```
Right?

Reply:
No, I am referring to the type of the lambda below this comment. I removed some comments from it explaining the calculation, since copying these comments tends to lead to two differently outdated versions.

Reply:
Ok, please correct